internal/service/cognitoidp/user.go

```diff
@@ -0,0 +1,565 @@
+package cognitoidp
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceUserCreate,
+		Read:   resourceUserRead,
+		Update: resourceUserUpdate,
+		Delete: resourceUserDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceUserImport,
+		},
+
+		// https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminCreateUser.html
+		Schema: map[string]*schema.Schema{
+			"client_metadata": {
+				Type:     schema.TypeMap,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			"creation_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"desired_delivery_mediums": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						cognitoidentityprovider.DeliveryMediumTypeSms,
+						cognitoidentityprovider.DeliveryMediumTypeEmail,
+					}, false),
+				},
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"force_alias_creation": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"last_modified_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"message_action": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					cognitoidentityprovider.MessageActionTypeResend,
+					cognitoidentityprovider.MessageActionTypeSuppress,
+				}, false),
+			},
+			"mfa_preference": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"sms_enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"software_token_enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"preferred_mfa": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				Computed: true,
+			},
+			"attributes": {
```

Comment:
I'm not sure how to make the `attributes` to be a block. According to the [Plugin Development guide](https://www.terraform.io/docs/extend/schemas/schema-types.html#typemap) next schema

```go
"tags": {
  Type:     schema.TypeMap,
  Elem: &schema.Schema{
    Type: schema.TypeString,
  },
},
```
should result in block argument:
```hcl
resource "example_compute_instance" "ex" {
  tags {
    env = "development"
    name = "example tag"
  }
}
```

But I'm getting attribute configuration:


```hcl
attributes = {
    bar            = "baz"
}
```